### DISCUSSION TITLE
use Hooks as useRouter() in _document.js

### DISCUSSION THREAD

**Discussion by Josehower** [vercel/next.js#43961] [Help]
I understand `_document.js` is a component that is only rendered on the server, but  I wonder if there is a way to make `useRouter` work there somehow. Always that I try to `useRouter` there I receive a https://nextjs.org/docs/messages/next-router-not-mounted. Are hooks only available for components that access the browser?

thanks in advance.

**Comment by ciruz** [comment]
With ``next 13`` and the ``/app`` directory all components are ``Server Components`` by default. When you try to use client hooks there, you need to define them as ``Client Components`` with `` `use client` ``.

I don't know which next version you use, or what you trying to achive, but you can also checkout ``this.props.__NEXT_DATA__`` in ``document.js``, for example you can access the page / query string with it. 

Greetings to vienna.

**Reply by Josehower** [reply]
Thanks for the answer,

I didn't know about `this.props.__NEXT_DATA__ `, good information.

After some further research, some hooks like `useEffect` or `useState` are not really useful  _document component, and I think if we need to render something or perform some code conditionaly based on the route, probably is better to put this code on _app instead.

Greetings to Großklein.
